### DISCUSSION TITLE
Redirect to external page

### DISCUSSION THREAD

**Discussion by cawa-93** [nuxt/nuxt#17497] [Questions]
I need to write a server middleware for the page that will check certain criteria and send a redirect to the external page.

How to do it? If I just call 
```ts
    navigateTo('https://github.com/nuxt/framework', {
      replace: true,
      redirectCode: 301,
    })
```
 then I am redirected to
```
http://localhost:3000/https://github.com/nuxt/framework
```

**Comment by wobsoriano** [comment]
You are looking for `sendRedirect`.

```ts
export default defineEventHandler((event) => {
  sendRedirect(event, 'https://github.com/nuxt/framework', 301)
})
```

**Reply by cawa-93** [reply]
But, can I use `sendRedirect` within page middleware?

**Reply by wobsoriano** [reply]
Oh, what you mean is you have a server middleware that returns the url you will redirect the user to from the page middleware?

**Reply by cawa-93** [reply]
I mean, I have a regular nuxt page that should to redirect the visitor. I want to create middleware for it. 
```vue
<script setup>
// pages/my-page.vue

definePageMeta({
  middleware: () => {
    // some checks and redirect if needed
    // Can I somehow use h3 sendRedirect here?
  }
})
</script>
```
And in my case, it is very important that in the case of an middleware runs on the server-side, the redirection header is returned immediately. Business logic does not allow you to wait for the page to load on the client to execute `location.href = ''`

At the moment, I see something like this workaround:
```vue
<script setup>
// pages/my-page.vue
definePageMeta({
  middleware: () => {
    // some checks and redirect if needed
    if (proccess.client) {
      location.href = 'https://github.com'
    } else {
      return navigateTo('/api/server-side-redirect?to=https://github.com') // server-side-redirect calls h3 sendRedirect 
    }
  }
})
```
